_posts/2016/2016-05-13-geecon.md

```diff
@@ -0,0 +1,86 @@
+<script async src="//platform.twitter.com/widgets.js" charset="utf-8"></script>
+
+## Daniel Sawano and Daniel Deogun - "Beyond lambdas - the aftermatch"
+
+The two Daniels ([@DanielSawano](https://twitter.com/DanielSawano), [@DanielDeogun](https://twitter.com/DanielDeogun)) focussed their talk on lambda expressions introduced in Java 8. In my team we love this language construct and use it extensively. The talk showed some common problems, smells and pitfalls when using lambdas:
+
+- things that are obvious for functional programmers can be hard to adapt to for java developers and you need to be very well  aware of what gets evaluated and when - so you better know the difference between `Optional.orElse` and `Optional.orElseGet`
+- the compact and fluent syntax of lambdas can hide issues where the separation of concerns is violated
+
+I liked the talk because the topic was specific and small so it was possible to dive rather deep. Also they presented it in a really nice way by illustrating their points with live-coding and telling the whole story as a dialogue between two developers.
+
+> <blockquote class="twitter-tweet" data-lang="de"><p lang="en" dir="ltr"><a href="https://twitter.com/DanielSawano">@DanielSawano</a> <a href="https://twitter.com/DanielDeogun">@DanielDeogun</a> great talk about pitfalls and best-practices with Java8 Lambdas - and fun too 💯 <a href="https://twitter.com/hashtag/geecon?src=hash">#geecon</a> <a href="https://t.co/gWpkvIozLK">pic.twitter.com/gWpkvIozLK</a></p>&mdash; Mathias D (@zaddo) <a href="https://twitter.com/zaddo/status/730666445688979456">12. Mai 2016</a></blockquote>
+<script async src="//platform.twitter.com/widgets.js" charset="utf-8"></script>
+
+## Julian Warszawski - "Greenfield Effect: Patterns for Effective Disaster Delivery"
+
+Julian Warszawski ([@hundredmondays](https://twitter.com/hundredmondays)) gave a very refreshing talk about the current state of our industry. He asked why developers so often start a greenfield project with best intentions and find themselves with a big ball of mud after some time that is harder and harder to maintain and delivers features slower and slower.
+
+One of the reasons he gave was technical debt and decisions taken by the wrong people. He stated that technical debt is a misleading term because the association to financial debt. Financial debt can easily be quantified and paid back in the future which is not the case for technical debt. And we better try to not pile up any because experience shows that it lasts forever and slows our work down.
```

Comment:
One of the reasons he gave > One of the reasons he presented
because the association to financial debt > because of the association with financial debt